internal/rpmmd/repository.go

```diff
@@ -652,6 +660,55 @@ func (r *rpmmdImpl) chainDepsolve(chains []chainPackageSet, repos []RepoConfig,
 	return dependencies, reply.Checksums, err
 }
 
+// DepsolvePackageSets takes a map of package sets chains, which should be depsolved as separate transactions,
+// a map of package sets (included and excluded), a list of common and package-set-specific repositories,
+// platform ID for modularity, architecture and release version. It returns a map of Package Specs, depsolved
+// in a chain or alone, as defined based on the provided arguments.
+func (r *rpmmdImpl) DepsolvePackageSets(
+	packageSetsChains map[string][]string,
```

Comment:
I think that this kind of methods taking 7 weird arguments should just take 1 struct but that's certainly something for a followup.

Reply:
I understand your feeling, but I opted in for 7 weird arguments instead of 1 weird structure. But I agree that we should follow-up and polish the whole `RPMMD` interface.

Reply:
I think these kinds of changes can be folded into #2537